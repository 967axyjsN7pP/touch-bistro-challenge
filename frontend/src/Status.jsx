import React from "react";

function Error(props) {
    return <p className="error">{props.message}</p>;
}

function Result(props) {
    return <p className="result">{props.message}</p>;
}

function Status(props) {
    if (props.errorMessage) {
        return <Error message={props.errorMessage} />;
    }
    else if (props.isFetching) {
        return <Result message="Fetching..." />;
    }
    else if (props.medianPrimes.length > 1) {
        return <Result message={`The median primes of ${props.limit} are ${props.medianPrimes.join(' and ')}`} />;
    }
    else if (props.medianPrimes.length === 1) {
        return <Result message={`The median prime of ${props.limit} is ${props.medianPrimes[0]}`} />;
    }

    return null;
}

export default Status;
