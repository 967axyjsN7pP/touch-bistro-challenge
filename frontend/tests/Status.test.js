import React from 'react';
import Status from '../src/Status.jsx';
import renderer from 'react-test-renderer';

test('it renders the fetching message when isFetching prop is true', () => {
    const component = renderer.create(
        <Status
            isFetching={true}
        />
    );
    expect(component.toJSON()).toMatchSnapshot()
});

test('it renders the error message when errorMessage prop is non-empty', () => {
    const component = renderer.create(
        <Status errorMessage="Something went wrong." />
    );
    expect(component.toJSON()).toMatchSnapshot();
});

test('it renders the singular-form of the success message when medianPrimes prop has a single element', () => {
    const component = renderer.create(
        <Status
            limit={18}
            medianPrimes={[7]}
        />
    );
    expect(component.toJSON()).toMatchSnapshot();
});

test('it renders the plural-form of the success message when medianPrimes prop has two elements', () => {
    const component = renderer.create(
        <Status
            limit={10}
            medianPrimes={[3, 5]}
        />
    );
    expect(component.toJSON()).toMatchSnapshot();
})
