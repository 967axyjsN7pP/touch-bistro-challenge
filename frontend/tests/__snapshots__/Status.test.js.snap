// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the error message when errorMessage prop is non-empty 1`] = `
<p
  className="error"
>
  Something went wrong.
</p>
`;

exports[`it renders the fetching message when isFetching prop is true 1`] = `
<p
  className="result"
>
  Fetching...
</p>
`;

exports[`it renders the plural-form of the success message when medianPrimes prop has two elements 1`] = `
<p
  className="result"
>
  The median primes of 10 are 3 and 5
</p>
`;

exports[`it renders the singular-form of the success message when medianPrimes prop has a single element 1`] = `
<p
  className="result"
>
  The median prime of 18 is 7
</p>
`;
